cdap-master/src/main/java/co/cask/cdap/master/environment/k8s/LogSaverServiceMain.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.master.environment.k8s;
+
+import ch.qos.logback.classic.LoggerContext;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.Appender;
+import co.cask.cdap.api.logging.AppenderContext;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.guice.LocalLocationModule;
+import co.cask.cdap.common.logging.LoggingContext;
+import co.cask.cdap.common.logging.ServiceLoggingContext;
+import co.cask.cdap.common.namespace.guice.NamespaceQueryAdminModule;
+import co.cask.cdap.data.runtime.DataSetsModules;
+import co.cask.cdap.data.runtime.SystemDatasetRuntimeModule;
+import co.cask.cdap.gateway.handlers.CommonHandlers;
+import co.cask.cdap.logging.appender.LogAppender;
+import co.cask.cdap.logging.appender.LogMessage;
+import co.cask.cdap.logging.framework.distributed.DistributedAppenderContext;
+import co.cask.cdap.logging.guice.LogQueryServerModule;
+import co.cask.cdap.logging.logbuffer.LogBufferService;
+import co.cask.cdap.logging.read.FileLogReader;
+import co.cask.cdap.logging.read.LogReader;
+import co.cask.cdap.logging.service.LogQueryService;
+import co.cask.cdap.logging.service.LogSaverStatusService;
+import co.cask.cdap.master.spi.environment.MasterEnvironment;
+import co.cask.cdap.master.spi.environment.MasterEnvironmentContext;
+import co.cask.cdap.messaging.guice.MessagingClientModule;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.security.auth.context.AuthenticationContextModules;
+import co.cask.cdap.security.authorization.AuthorizationEnforcementModule;
+import co.cask.cdap.security.impersonation.CurrentUGIProvider;
+import co.cask.cdap.security.impersonation.UGIProvider;
+import co.cask.http.HttpHandler;
+import com.google.common.util.concurrent.Service;
+import com.google.inject.AbstractModule;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import com.google.inject.PrivateModule;
+import com.google.inject.Scopes;
+import com.google.inject.multibindings.Multibinder;
+import com.google.inject.name.Names;
+import org.slf4j.ILoggerFactory;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * Main class for running log saver and log query service in Kubernetes.
+ */
+public class LogSaverServiceMain extends AbstractServiceMain {
```

Comment:
Call this `LogsServiceMain`, as it does both saving and serving of logs.

Reply:
Using `LogServiceMain` as we do not use plural in classnames.